addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4438,6 +4438,15 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
  * Controls image width and quality.
  */
 registry.ImageOptimize = ImageHandlerOption.extend({
+
+    willStart() {
+        const img = this._getImg();
+        if (img.dataset.shape && img.dataset.originalMimetype) {
+            img.dataset.mimetype = img.dataset.originalMimetype;
+        }
```

Comment:
The original mimetype is used for quality options. I also have to set the mimetype of the file to svg otherwise the image / shape will not load / be save properly.